packages/core/src/components/mediaDeviceSelect.ts

```diff
@@ -17,42 +16,17 @@ export function setupDeviceSelector(kind: MediaDeviceKind, room?: Room) {
   const activeDeviceSubject = new BehaviorSubject<string | undefined>(undefined);
 
   const activeDeviceObservable = room
-    ? observeParticipantMedia(room.localParticipant).pipe(
-        map((participantMedia) => {
-          let localTrack: LocalAudioTrack | LocalVideoTrack | undefined;
-          switch (kind) {
-            case 'videoinput':
-              localTrack = participantMedia.cameraTrack?.track as LocalAudioTrack;
-              break;
-            case 'audioinput':
-              localTrack = participantMedia.microphoneTrack?.track as LocalVideoTrack;
-              break;
-            default:
-              localTrack = undefined;
-              break;
-          }
-          return localTrack?.mediaStreamTrack.getSettings()?.deviceId;
-        }),
-        mergeWith(activeDeviceSubject),
-      )
+    ? createActiveDeviceObservable(room, kind).pipe(mergeWith(activeDeviceSubject))
     : activeDeviceSubject.asObservable();
 
   const setActiveMediaDevice = async (id: string, options: SetMediaDeviceOptions = {}) => {
     if (room) {
       log.debug(`Switching active device of kind "${kind}" with id ${id}.`);
       await room.switchActiveDevice(kind, id, options.exact);
       let actualDeviceId: string | undefined = id;
-      if (kind === 'videoinput') {
-        actualDeviceId = await room.localParticipant
-          .getTrack(Track.Source.Camera)
-          ?.track?.getDeviceId();
-      } else if (kind === 'audioinput') {
-        actualDeviceId = await room.localParticipant
-          .getTrack(Track.Source.Microphone)
-          ?.track?.getDeviceId();
-      }
+
       if (actualDeviceId !== id && id !== 'default') {
-        log.warn(`Failed to select the desired device. Desired: ${id}. Actual: ${actualDeviceId}`);
+        actualDeviceId = room.getActiveDevice(kind);
```

Comment:
should we have a warn/info log still if the check fails of actualDeviceId against the one that was passed in by the user?

Reply:
Sounds good!